creator-node/src/snapbackSM/syncHistoryAggregator.js

```diff
@@ -0,0 +1,143 @@
+const moment = require('moment')
+
+const redisClient = require('../redis')
+const { logger: genericLogger } = require('../logging')
+const config = require('../config')
+
+const CREATOR_NODE_ENDPOINT = config.get('creatorNodeEndpoint')
+
+const SYNC_STATES = Object.freeze({
+  triggered: 'triggered',
+  success: 'success',
+  fail: 'fail'
+})
+
+// Make key expire in 7 days in seconds
+const EXPIRATION = 7 /* days */ * 24 /* hr */ * 60 /* min */ * 60 /* s */
+
+/**
+ * Class intended to:
+ * - Record the number of the current day's triggered, successful, and failed sync attempts
+ * - Record the timestamp of the most recent successful and failed sync
+ */
+
+// Note: When recording a sync 'success' or 'fail', it implies that the secondary was able to respectfully
+// successfully or unsuccessfully take the primary's exported data, and update its own state
+
+class SyncHistoryAggregator {
+  static async recordSyncSuccess (logContext) {
+    await SyncHistoryAggregator.recordSyncData({
+      state: SYNC_STATES.success,
+      timeOfEvent: moment().format('MM-DD-YYYYTHH:MM:SS:sss'),
+      logContext
+    })
+  }
+
+  static async recordSyncFail (logContext) {
+    await SyncHistoryAggregator.recordSyncData({
+      state: SYNC_STATES.fail,
+      timeOfEvent: moment().format('MM-DD-YYYYTHH:MM:SS:sss'),
+      logContext
+    })
+  }
+
+  static async recordSyncData ({ state, timeOfEvent, logContext }) {
+    const logger = genericLogger.child(logContext)
+
+    try {
+      // Update aggregate sync data
+      const aggregateSyncKey = SyncHistoryAggregator.getAggregateSyncKey(CREATOR_NODE_ENDPOINT)
+      const existingAggregateSyncEntry = await redisClient.get(aggregateSyncKey)
+      if (!existingAggregateSyncEntry) {
+      // Init aggregate sync data
+        await redisClient.set(aggregateSyncKey,
+          JSON.stringify({
+            triggered: 0,
```

Comment:
yeah this is a good call out. the reason I didn't originally go with this was because I thought it would be more convenient to manage all the aggregate and latest sync data under 1 key. this option is valid as well, as it isn't expensive to add to redis. will update using incr